src/util/settings.cpp

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <util/settings.h>
+
+#include <univalue.h>
+
+namespace util {
+namespace {
+
+struct Source {
+    SettingsSpan span;
+    bool forced = false;
+    bool config_file = false;
+    bool config_file_top_level = false;
+
+    Source(SettingsSpan span) : span(span) {}
+    Source& SetForced() { forced = true; return *this; }
+    Source& SetConfigFile(bool top_level) { config_file = true; config_file_top_level = top_level; return *this; }
+};
+
+template <typename Fn>
+static void MergeSettings(const Settings& settings, const std::string& section, const std::string& name, Fn&& fn)
+{
+    if (auto* value = FindKey(settings.forced_settings, name)) {
+        fn(Source(SettingsSpan(value)).SetForced());
```

Comment:
nit: any reason why you don't dereference compare to others?

Reply:
> nit: any reason why you don't dereference compare to others?

sss